ecosystem/typescript/sdk_v2/src/utils/memoize.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ * The global cache Map
+ */
+const cache = new Map<string, { value: any; timestamp: number }>();
+
+/**
+ * A memoize high order function to cache async function response
+ *
+ * @param func An async function to cache the result of
+ * @param key The provided cache key
+ * @param ttlMs time-to-live in milliseconds for cached data
+ * @returns
+ */
+export function memoizeAsync<T>(
+  func: (...args: any[]) => Promise<T>,
+  key: string,
+  ttlMs?: number,
+): (...args: any[]) => Promise<T> {
+  return async (...args: any[]) => {
+    // Check if the cached result exists and is within TTL
+    if (cache.has(key)) {
```

Comment:
Noob q - Is this global cache for the whole sdk? 

If I have two functions
```
function getA(value: string): string
function getB(value: string): string
```

Is it possible there might be collision if I am passing the same value e.g `test` to both function that are cached?

Reply:
well yes, but it shouldn't happen as we decide what the cache key will be, i.e `getAccountModules_${args.accountAddress}`. This is different than what it is today as the decorator assumes the cache key should be the first function argument.